File: app/src/main/java/com/pos/hardware/connection/demo/ui/BaseAppCompatActivity.java
package com.pos.hardware.connection.demo.ui;

import android.widget.Toast;

import androidx.annotation.StringRes;
import androidx.appcompat.app.AppCompatActivity;
/**
 * @author: Dadong
 * @date: 2024/11/21
 */


public class BaseAppCompatActivity extends AppCompatActivity {

    public void showToast(String text) {
        runOnUiThread(() -> Toast.makeText(BaseAppCompatActivity.this, text, Toast.LENGTH_SHORT).show());
    }

    public void showToast(@StringRes int resId) {
        runOnUiThread(() -> Toast.makeText(BaseAppCompatActivity.this, resId, Toast.LENGTH_SHORT).show());
    }
}

File: app/src/main/java/com/pos/hardware/connection/demo/ui/BaseAppCompatActivity.kt
package com.pos.hardware.connection.demo.ui

import android.widget.Toast
import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatActivity

open class BaseAppCompatActivity : AppCompatActivity() {

    fun showToast(text: String) {
        runOnUiThread { Toast.makeText(this, text, Toast.LENGTH_SHORT).show() }
    }

    fun showToast(@StringRes resId: Int) {
        runOnUiThread { Toast.makeText(this, resId, Toast.LENGTH_SHORT).show() }
    }

}